packages/calypso-e2e/src/lib/blocks/block-flows/ai-assistant.ts

```diff
@@ -0,0 +1,158 @@
+	}
+
+	blockSidebarName = 'AI Assistant (Experimental)';
+	blockEditorSelector = 'div[aria-label="Block: AI Assistant (Experimental)"]';
+
+	/**
+	 * Configure the block in the editor with the configuration data from the constructor
+	 *
+	 * @param {EditorContext} context The current context for the editor at the point of
+	 * test execution.
+	 */
+	async configure( context: EditorContext ): Promise< void > {
+		const editorCanvas = await context.editorPage.getEditorCanvas();
+		const block = editorCanvas.getByRole( 'document', {
+			name: `Block: ${ this.blockSidebarName }`,
+		} );
+
+		await this.enterInput( block );
+		await this.clickButtonOnBlock( block, 'Send' );
+		await this.waitForQuery( editorCanvas );
+
+		if ( this.configurationData.tone ) {
+			await this.clickToolbarButton( 'Change tone', this.configurationData.tone, context );
+			await this.waitForQuery( editorCanvas );
+		}
+
+		if ( this.configurationData.improve ) {
+			await this.clickToolbarButton( 'Improve', this.configurationData.improve, context );
+			await this.waitForQuery( editorCanvas );
+		}
+
+		await this.clickButtonOnBlock( block, 'Accept' );
+		await block.waitFor( { state: 'detached' } );
+		this.generatedText = await editorCanvas.locator( '.is-selected' ).innerText();
+	}
+
+	/**
+	 * Clicks on the toolbar button, then in the dropdown that appears, clicks an option.
+	 *
+	 * @param {string} buttonName Name of the button to click.
+	 * @param {string} optionName Name of the option in the dropdown to click.
+	 * @param {EditorContext} context The current context for the editor at the point of
+	 * test execution.
+	 */
+	private async clickToolbarButton(
+		buttonName: string,
+		optionName: string,
+		context: EditorContext
+	) {
+		// Block-based and non-block-based themes strike again.
+		// For block-based themes (eg. Twenty Twenty-Three) a given block's toolbar
+		// lives "outside" of the DOM tree returned by `getEditorCanvas`.
+		// For non-block-based themes (eg. Twenty Twenty-One) a block's toolbar
+		// lives in the same DOM tree as the locator returned by `getEditorCanvas`.
+		// However, the locator returned by `getEditorParent` has the toolbar inside
+		// the DOM tree in both cases.
+		const editorParent = await context.editorPage.getEditorParent();
+
+		const toolbar = editorParent.getByRole( 'toolbar', { name: 'Block tools' } );
+		await toolbar.waitFor();
+		// Yes, yuck, I know. But Playwright often tries to act on the buttons sooner
+		// than it should, so we need to wait for the bounding box of the element here
+		// to be stable before attempting to click on the toolbar.
+		const elementHandle = await toolbar.elementHandle();
+		await elementHandle?.waitForElementState( 'stable' );
+
+		const button = toolbar.getByRole( 'button', { name: buttonName } );
+		await button.waitFor();
+		await button.click();
+
+		const option = editorParent.getByRole( 'menuitem', { name: optionName } );
+		await option.waitFor();
+		await option.click();
```

Comment:
**Suggestion (non-blocking):** We have component POM classes for these already! ([Block toolbar](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/components/editor-block-toolbar-component.ts) and [popover menu](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-e2e/src/lib/components/editor-popover-menu-component.ts).) What do you think about extending/enhancing and using those, to keep things DRYish?